### ISSUE TITLE
transitionTo produces duplicate <Router> element

### ISSUE THREAD

**Issue by jabbrwcky** [rackt/react-router#1663]
This might just be my lack of understanding, but I have the following use-case:

I have an application that can authenticate users via (enterprise) github to create an JWT token for my backend.

The authentication process consists of several redirections for the user that finally redirects the browser back to my application using some URL like `http://localhost:8080/login/callback?token=<backend api token>`.

The idea is to grab the API token, store it in a store/localStorage and to redirect the user to the main page. The application uses react-router 0.13.3 at the moment.

My Login component uses the Navigation mixin and calls 
`this.transitionTo('main')` resulting in a loads of React errors complaining that the root element has been removed and theChrome React Console plugin displaying the following page structure:

```
<TopLevel>
  <Router path='/login/callback?='...'>
    ... (login component)
  </Router>
  <Router path='/home' >
    ... (authenticated content)
  </Router>
</TopLevel>
```

Somewhat shortened routes definition:
```
  <Route handler={App} >
    <Route handler={LoggedOut}>
      <DefaultRoute handler={Welcome} />
      <Route name ='login' path='login' handler={Login}>
        <Route name = ":method" handler={Login} />
      </Route>
    </Route>
    <Route handler={LoggedIn}>
      <Route path='home' name='main' handler={Main} />
    </Route>
    <NotFoundRoute handler={NotFound} />
  </Route>
```

I am puzzled where the second Router comes from (Could it be the self-nesting Login?). 

As all googling, stack-overflowing yielded no result, I try the issue.

**Comment by ryanflorence** [comment]
sorry, I have no idea what this could be caused by, let me know if you find out anything else, but I don't think there's anything actionable for React Router here. If you discover otherwise, let us know.

**Comment by jabbrwcky** [comment]
I found that the react rendering process is extremely sensitive to any kind of redirects before finishing the render phase, i.e. triggering a redirect from an action (by passing the router context via action arguments).

Redirects triggered by user actions (e.g. clicking on a link are ok, as are redirects in the transition callbacks). I resolved the issue by doing "automatic" redirects in the `componentDidUpdate` hook of my top level App component.
